### PR TITLE
Constexpr next

### PR DESCRIPTION
**Pull request by mborland**
Implements constexpr: `nextafter`, `nextafterf`, `nextafterl`, `nexttoward`, `nexttowardf`, and `nexttowardl`. Once this and #734 are complete [all functions pending standardization ](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p0533r9.pdf)will be constexpr.

### CODE REVIEW

**Comment by mborland** [comment]
@NAThompson Can you review this? It is almost a 1:1 copy of the existing nextafter implementation with standard library cmath functions being replaced by our ccmath functions. The results match exactly which makes me feel good about the other functions.

File: test/ccmath_next_test.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    // test_1 has already passed through static_asserts so we know it was calculated at compile time
+    // rather than farming out to std at run time.
+    const T existing_test_1 = boost::math::nextafter(T(1), T(1.5));
+    CHECK_EQUAL(test_1, existing_test_1);
+}
+
+int main(void)
+{
+    test_next<float>();
+    test_next<double>();
```
Comment (NAThompson) [inline test/ccmath_next_test.cpp:49]: @mborland : Is there a way to test for float16?

**Comment by NAThompson** [comment]
@mborland : Looks good; make sure to squash merge with a nice commit message.

File: test/ccmath_next_test.cpp (hunk 1 above)
Comment (mborland) [reply test/ccmath_next_test.cpp:49]: I don't see any macros in cstdfloat that allow a portable way of checking that `float16_t` is available. I believe that fixed width floating points are being added in C++23 with the reference implementation coming from Boost.Math.

File: include/boost/math/ccmath/next.hpp [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+        diff = detail::get_smallest_value<T>();
+    }
+    return val + diff;
+}
+
+//
+// Special version for some base other than 2:
+//
+template <typename T>
+constexpr T float_next_imp(const T& val, const std::false_type&)
```
Comment (jzmaddock) [inline include/boost/math/ccmath/next.hpp:199]: Do we actually need to support bases other than 2 in the constexpr case?  I would be inclined to remove this and static_assert that the radix is 2.

**Pull request merged**